packages/amplify_storage_s3/example/integration_test/main_test.dart

```diff
@@ -0,0 +1,121 @@
+
+const exampleFileName = 'example_file.txt';
+
+void main() async {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('amplify_storage_s3', () {
+    // options used for all tests
+    S3ListOptions listOptions =
+        S3ListOptions(accessLevel: StorageAccessLevel.guest);
+    late String? lastUploadedKey;
+
+    // Returns a text file to use for testing, writing if it does not exist.
+    Future<File> getTemporaryFile() async {
+      final path = '${(await getTemporaryDirectory()).path}/$exampleFileName';
+      final file = File(path);
+      if (!(await file.exists())) {
+        await file.writeAsString('Upload me to s3 to see if I work.');
+      }
+      return file;
+    }
+
+    Future<int> getCountFromListFiles() async {
+      final result = await Amplify.Storage.list(options: listOptions);
+      return result.items.length;
+    }
+
+    Future<void> deleteAllGuestFiles() async {
+      final result = await Amplify.Storage.list(options: listOptions);
+      for (StorageItem item in result.items) {
+        await Amplify.Storage.remove(key: item.key);
+      }
+    }
+
+    setUpAll(() async {
+      Amplify.addPlugins([AmplifyAuthCognito(), AmplifyStorageS3()]);
+      await Amplify.configure(amplifyconfig);
+
+      await deleteAllGuestFiles();
+    });
+
+    tearDownAll(() async {
+      await deleteAllGuestFiles();
+    });
+
+    testWidgets('should upload a file', (WidgetTester tester) async {
+      final file = await getTemporaryFile();
+      final initialCount = await getCountFromListFiles();
+
+      final key = '$exampleFileName${new DateTime.now().toString()}';
+      Map<String, String> metadata = <String, String>{};
+      metadata['name'] = exampleFileName;
+      metadata['desc'] = 'A test file';
+      S3UploadFileOptions options = S3UploadFileOptions(
+          accessLevel: StorageAccessLevel.guest, metadata: metadata);
+      final result = await Amplify.Storage.uploadFile(
+          key: key, local: file, options: options);
+      lastUploadedKey = result.key;
+      expect(lastUploadedKey, key);
+
+      final finalCount = await getCountFromListFiles();
+      expect(initialCount + 1, finalCount);
+    });
+
+    testWidgets('should list files and get expected attributes',
+        (WidgetTester tester) async {
+      if (lastUploadedKey == null) {
+        fail('No uploaded file to verify.');
+      }
+      final result = await Amplify.Storage.list(options: listOptions);
+      expect(result.items.length, greaterThan(0));
+      final uploadedStorageItem =
+          result.items.firstWhere((element) => element.key == lastUploadedKey);
+      expect(uploadedStorageItem.key, lastUploadedKey);
+      expect(uploadedStorageItem.lastModified?.day,
+          new DateTime.now().day); // was uploaded today
+      expect(uploadedStorageItem.eTag?.isNotEmpty, true);
+      expect(uploadedStorageItem.size, greaterThan(0));
```

Comment:
Would it be good to check the metadata items here as well? Like the `name` and `desc` fields set when uploading.

Reply:
Yes, would be helpful, will add.

Reply:
Discussed offline, but the metadata is not available on StorageItem so not possible do this at the moment.